### ISSUE TITLE
Still working?

### ISSUE THREAD

**Issue by opb** [fotherja/EV_Charger#6]
Hi James. I’ve been eying up your project for a while and even got some PCBs printed. I just wanted to double check that it’s all working as expected a year or more after it looks like this was completed. Any additional thoughts since the project was last updated?

Cheers,
Olly

**Comment by fotherja** [comment]
Hi Olly,
I've been using the charger a few times a week since this repo was created. It's worked flawlessly. I think the last time I opened up the enclosure was over a year ago and that was to add the watch dog timer and fix a software bug that was identified early on. But the code here has those updates. 

I've never hankered after any smart features because I can set charge times using my car's phone app and my electricity is monitored by my smart meter anyway.

I can't say that all my projects have proven to be as reliable as this one. It worked through last year's scorching summer and this winters -10oC nights. So I'm genuinely very pleased with it.

I have noticed on the pilot pin of the charging plug a very small amount of blue (I guess copper) corrosion. It's barely noticeable even after a year and I don't think it'll ever cause a problem. plugging/unplugging keeps the contacts clean. I was wondering if this could be because 12v DC is on the pilot pin continuously. I'm not going to do anything about it - since I noticed it I always hang the charging plug down so water will never pool inside the plug (it's all outside). If I had to think of an improvement on this project I'd look into making (hardware and software changes) so that in the idle state, the hardware only occasionally applies 12v to the pilot pin to see whether a car is connected and if not keeps the pilot pin at 0v. If that makes sense?

**Comment by opb** [comment]
Hi James

Thanks for the response. Great to hear it’s still working flawlessly! I was contemplating making a a couple of adjustments to get a networked Arduino in there, and have it communicate state or error state to Home Assistant. However, I have a networked 40A relay with power monitoring that I’ll put into the enclosure for turning overall power on or off. I think trying to modify the code and PCB would just mean I’ll never get it completed!

Plus, if there is an issue with the 12v pulse running constantly, hopefully that would be resolved with the ability to turn on and off to a schedule with Home Assistant. Sadly my car has no app or any smart features. 

Last question, sorry - how does the PWM value of 350 translate to the 30-32A needed for the 7.4kW you mention in the code comments? The linked docs in your PDF indicate a 50-55% duty cycle for this current, so I’m keen to understand how the value in the code maps to the percentage. If you have any insight, it would be much appreciated! 

Thanks

**Comment by fotherja** [comment]
Hi Olly,
Regarding the PWM value of 350. Good point. This was trial and error for me. I tried a few values and this resulted in a 7.4kW charge rate. Can't really say how it translates although when I scoped the PWM signal being generated there is quite a noticeable skew rate from the op-amp output. Perhaps this is why it doesn't translate accurately. A faster op-amp might help this. I've never charged any other car than my Kia ENiro so I don't know if different cars would read the PWM signal differently because of the slew and charge at slightly different rates.
